### PR TITLE
Chapter 8 Basic login

### PR DESCRIPTION
**Pull request by tainqa-0955**

### CODE REVIEW

**Comment by tainqa-0955** [comment]
Screenshot from 2019-09-19 19-05-35

**Comment by tainqa-0955** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id] if session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```
Comment (manhbnt-0960) [inline app/helpers/sessions_helper.rb]:
```suggestion
    current_user.present?
```

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def log_out
+    session.delete(:user_id)
```
Comment (manhbnt-0960) [inline app/helpers/sessions_helper.rb]:
```suggestion
    session.delete :user_id
```

File: app/models/user.rb [hunk 3]
```diff
@@ -10,4 +10,13 @@ class User < ApplicationRecord
   validates :password, presence: true,
     length: {minimum: Settings.validates[:minimum_password]}
   has_secure_password
+
+  def self.digest string
```
Comment (manhbnt-0960) [inline app/models/user.rb]: cho vào class << self đi e

**Comment by manhbnt-0960** [comment]
commented

**Comment by tainqa-0955** [comment]
ready

**Comment by chienpv-0644** [comment]
merged

**Pull request merged**